c/src/connection.rs

```diff
@@ -17,56 +17,88 @@
+/// @param driver_lang The language of the driver connecting to the server
+#[no_mangle]
+pub extern "C" fn driver_open_cloud(
+    addresses: *const *const c_char,
+    credential: *const Credential,
+    connection_settings: *const ConnectionSettings,
+    driver_lang: *const c_char,
+) -> *mut TypeDBDriver {
+    // TODO: Add a separate entry point for C with a provided "c" driver_lang!
+    let addresses: Vec<&str> = string_array_view(addresses).collect();
+    try_release(TypeDBDriver::new_cloud_with_description(
+        &addresses,
+        borrow(credential).clone(),
+        borrow(connection_settings).clone(),
+        string_view(driver_lang),
+    ))
+}
 
-///// Open a TypeDB Driver to TypeDB Cloud server(s), using provided address translation, with
-///// the provided credential.
-/////
-///// @param public_addresses A null-terminated array holding the address(es) of the TypeDB server(s)
-///// the driver will connect to. This array <i>must</i> have the same length as <code>advertised_addresses</code>
-///// @param private_addresses A null-terminated array holding the address(es) the TypeDB server(s)
-///// are configured to advertise
-///// @param credential The <code>Credential</code> to connect with
-// #[no_mangle]
-// pub extern "C" fn driver_open_cloud_translated(
-//     public_addresses: *const *const c_char,
-//     private_addresses: *const *const c_char,
-//     credential: *const Credential,
-// ) -> *mut TypeDBDriver {
-//     let addresses = string_array_view(public_addresses).zip_eq(string_array_view(private_addresses)).collect();
-//     try_release(TypeDBDriver::new_cloud_with_translation(addresses, borrow(credential).clone()))
-// }
+/// Open a TypeDB Driver to TypeDB Cloud server(s), using provided address translation.
+///
+/// @param public_addresses A null-terminated array holding the address(es) of the TypeDB server(s)
+/// the driver will connect to. This array <i>must</i> have the same length as <code>advertised_addresses</code>
+/// @param private_addresses A null-terminated array holding the address(es) the TypeDB server(s)
+/// are configured to advertise
+/// @param credential The <code>Credential</code> to connect with
+/// @param connection_settings The <code>ConnectionSettings</code> to connect with
+/// @param driver_lang The language of the driver connecting to the server
+#[no_mangle]
+pub extern "C" fn driver_open_cloud_translated(
+    public_addresses: *const *const c_char,
+    private_addresses: *const *const c_char,
+    credential: *const Credential,
+    connection_settings: *const ConnectionSettings,
+    driver_lang: *const c_char,
+) -> *mut TypeDBDriver {
+    // TODO: Add a separate entry point for C with a provided "c" driver_lang!
+    let addresses = string_array_view(public_addresses).zip_eq(string_array_view(private_addresses)).collect();
+    try_release(TypeDBDriver::new_cloud_with_translation_with_description(
```

Comment:
Don't like this copypaste with `with_..._with_...`, can be refactored later when the actual implementation starts.